### PR TITLE
fix: support rendering nested pyproject.toml outputs to stdout

### LINKED ISSUE #85: Rendering a nested pyproject.toml file to stdout is broken
Discovered this today while working on https://github.com/rapidsai/cudf/pull/15245.

On the latest version of `dependency-file-generator` (v1.13.4), rendering a nested `pyproject.toml` to stdout fails.

## Reproducible Example

```shell
git clone git@github.com:rapidsai/cudf.git
cd cudf
git checkout b810113d

rapids-dependency-file-generator \
  --output pyproject \
  --file-key py_run_cudf \
  --matrix "cuda=12.2"
```

Fails like this:

```text
Traceback (most recent call last):
  File "/Users/jlamb/miniforge3/envs/rapids-dev/lib/python3.10/site-packages/rapids_dependency_file_generator/_rapids_dependency_file_generator.py", line 183, in make_dependency_file
    table = table[section]
  File "/Users/jlamb/miniforge3/envs/rapids-dev/lib/python3.10/site-packages/tomlkit/container.py", line 624, in __getitem__
    item = self.item(key)
  File "/Users/jlamb/miniforge3/envs/rapids-dev/lib/python3.10/site-packages/tomlkit/container.py", line 466, in item
    raise NonExistentKey(key)
tomlkit.exceptions.NonExistentKey: 'Key "project" does not exist.'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jlamb/miniforge3/envs/rapids-dev/bin/rapids-dependency-file-generator", line 10, in <module>
    sys.exit(main())
  File "/Users/jlamb/miniforge3/envs/rapids-dev/lib/python3.10/site-packages/rapids_dependency_file_generator/_cli.py", line 151, in main
    make_dependency_files(parsed_config, file_keys, output, matrix, args.prepend_channels, to_stdout)
  File "/Users/jlamb/miniforge3/envs/rapids-dev/lib/python3.10/site-packages/rapids_dependency_file_generator/_rapids_dependency_file_generator.py", line 419, in make_dependency_files
    contents = make_dependency_file(
  File "/Users/jlamb/miniforge3/envs/rapids-dev/lib/python3.10/site-packages/rapids_dependency_file_generator/_rapids_dependency_file_generator.py", line 188, in make_dependency_file
    if not table.is_super_table():
AttributeError: 'TOMLDocument' object has no attribute 'is_super_table'
```

## Notes

I believe the specific issue is right here:

https://github.com/rapidsai/dependency-file-generator/blob/ec3d043e7ab0abb5acb111ebdeea0c72b691d0a5/src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py#L418

### PR DESCRIPTION
**Pull request by jameslamb**
Fixes #85.

Contributes to https://github.com/rapidsai/build-planning/issues/31 (via making debugging those changes easier).

#85 contains a reproducible example and details of the problem this solves. In short:

* when rendering to stdout, `dependency-file-generator` currently defaults to looking for the file to overwrite **in the current working directory**
* that leads to it use, say, `./pyproject.toml` as the source for an invocation that should be handling e.g. `./python/cudf/pyproject.toml`

This proposes the following:

* modifies `make_dependency_files()` to always `conda_dir` / `pyproject_dir` / `requirements_dir` as a source of the file to load / possibly overwrite
* fixes some outdated stuff in `get_output_dir()` docstring
* changes `get_output_dir()` to use keyword-only arguments
* adds a test covering this functionality

### How I tested this

Installed the version from `main` on this branch.

```shell
git checkout main -- src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py
pip install --ignore-installed .
```

Confirmed that this new unit test fails on `main`, and is reading in the root-level `pyproject.toml` instead of the nested one.

```shell
pytest tests/test_rapids_dependency_file_generator.py
```

Confirmed that it fails with the same error as #85 on the new test project added here.

```shell
cd ./tests/examples/nested-pyproject
rapids-dependency-file-generator \
   --output pyproject \
   --file-key sparkly_unicorn \
   --matrix "cuda=100.1"

# tomlkit.exceptions.NonExistentKey: 'Key "project" does not exist.'
```

Tested on `cudf`'s actual `dependencies.yaml` on `branch-24.06`.

```shell
rapids-dependency-file-generator \
  --output pyproject \
  --file-key py_run_cudf \
  --matrix "cuda=12.2"
```

### CODE REVIEW

File: tests/examples/nested-pyproject/dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+files:
+  sparkly_unicorn:
```
Comment (KyleFromNVIDIA) [inline tests/examples/nested-pyproject/dependencies.yaml:2]: ✨🦄😄

**Review by KyleFromNVIDIA — APPROVED** [review]
+1. Thanks for fixing this!

File: tests/examples/nested-pyproject/dependencies.yaml (hunk 1 above)
Comment (jameslamb) [reply tests/examples/nested-pyproject/dependencies.yaml:2]: ha this is one of my favorite things to use to make it absolutely clear that something is user-controlled arbitrary input and not something that has special meaning to the library

Comment (KyleFromNVIDIA) [reply tests/examples/nested-pyproject/dependencies.yaml:2]: This is making me think of PJReddie's website: https://pjreddie.com/

**Review by ajschmidt8 — APPROVED** [review]
:firefighter:

**Pull request merged**